transports/tcp/src/lib.rs

```diff
@@ -327,6 +345,16 @@ where
         if let Some(nodelay) = self.nodelay {
             socket.set_nodelay(nodelay)?;
         }
+        if let Some(size) = self.send_buffer_size {
+            socket.set_send_buffer_size(size)?;
+            let actual = socket.send_buffer_size()?;
```

Comment:
Not sure how expensive this call is. Might be worth only doing it when debug or higher is enabled.